src/App/View/LinkedText.purs

```diff
@@ -2,63 +2,94 @@ module App.View.LinkedText where
 
 import Prelude
 
-import App.Util (class Reflect, SelState, Selectable, 𝕊, from)
+import App.Util (class Reflect, Attrs, SelState, Selectable, 𝕊, classes, from, isPersistent, isPrimary, isSecondary, isTransient)
 import App.Util.Selector (linkedText, listElement, ViewSelSetter)
-import App.View.Util (class Drawable, Renderer, selListener, uiHelpers)
-import Data.Either (Either(..))
-import Data.Tuple (Tuple, fst)
-import Primitive (Explanation(..), linkedTextEntry, string, unpack)
-import Util (type (+), (×))
+import App.View.Util (class Drawable, class Drawable2, draw', registerMouseListeners, selListener, uiHelpers)
+import App.View.Util.D3 (ElementType(..), create, datum, selectAll, setDatum, setStyles, setText)
+import App.View.Util.D3 as D3
+import Bind ((↦))
+import Data.Array (length)
+import Data.Foldable (for_)
+import Data.FoldableWithIndex (forWithIndex_)
+import Data.Tuple (fst, snd)
+import Effect (Effect)
+import Effect.Class.Console (logShow)
+import Effect.Console (log)
+import Primitive (string, unpack)
+import Util ((!))
 import Val (Val)
+import Web.Event.EventTarget (EventListener)
 
-foreign import drawLinkedText :: LinkedTextHelpers -> Renderer LinkedText
+newtype LinkedText = LinkedText (Array (Selectable String))
 
-type LinkedTextHelpers =
-   { explanation :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> String
-   , contents :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> String
-   , accessAnn :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> SelState 𝕊
-   }
+contents :: Selectable String -> String
+contents = fst
 
-newtype LinkedText = LinkedText (Array (Selectable String + Selectable (Explanation (SelState 𝕊))))
-
-drawLinkedText' :: Renderer LinkedText
-drawLinkedText' = drawLinkedText linkedTextHelpers
-
-linkedTextHelpers :: LinkedTextHelpers
-linkedTextHelpers =
-   { explanation
-   , contents
-   , accessAnn
-   }
-   where
-   explanation :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> String
-   explanation = case _ of
-      Left s -> fst s
-      Right (Explanation _ expl _ × _) -> expl
-
-   contents :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> String
-   contents = case _ of
-      Left s -> fst s
-      Right (Explanation _ _ v × _) -> fst (unpack string v)
-
-   accessAnn :: Selectable String + Selectable (Explanation (SelState 𝕊)) -> SelState 𝕊
-   accessAnn = case _ of
-      Left (_ × α) -> α
-      Right (Explanation α _ _ × _) -> α
+accessAnn :: Selectable String -> SelState 𝕊
```

Comment:
Move to same module as `Selectable` and rename to `selState` (unless already taken, in which case `sel` might do).